### DISCUSSION TITLE
Can i rearrange the order of inputs in my model?

### DISCUSSION THREAD

**Discussion by Achidago** [SERVIR-WA/GALUP#97] [Q&A] [answered]
I realize the order of inputs in the model i created differs from the example shown.
For intance,
Soil PH (5-15 cm) comes before Soil PH (0-5 cm)

My Model
![My_Model](https://user-images.githubusercontent.com/85199074/124841699-5da60700-df7d-11eb-8bb2-b40942f03ff0.JPG)

GALUP Model
![GALUP_Model](https://user-images.githubusercontent.com/85199074/124841708-672f6f00-df7d-11eb-838b-84dc161084f7.JPG)

**Comment by chjch** [comment]
> I realize the order of inputs in the model i created differs from the example shown.
> For intance,
> Soil PH (5-15 cm) comes before Soil PH (0-5 cm)
> 
> My Model
> [My_Model.pdf](https://github.com/SERVIR-WA/GALUP/files/6780299/My_Model.pdf)
> 
> GALUP Model
> [GALUP_Model.pdf](https://github.com/SERVIR-WA/GALUP/files/6780300/GALUP_Model.pdf)

Hi Silas,

Please use image formats here in the discussion channel if possible. Please replace your two pdf files above. It will be much clearer.

Changjie

**Reply by Achidago** [reply]
Done

**Comment by chjch** [comment, marked as answer]
Hi Silas,

You asked an **important** question. We addressed this issue in [Exercise 1](https://github.com/SERVIR-WA/GALUP/blob/master/training/1_lu/exercises/m3_exercise1.md#31-build-the-model)'s instruction. A similar question was asked on GIS StackExchange, and a [solution](https://gis.stackexchange.com/a/316901/83211) was posted. 

The issue is related to the naming convention you adopted for the inputs. Graphical Modeler **automatically** rearranges the order of inputs based on the input name's first character in the [ASCII-code order](https://en.wikipedia.org/wiki/ASCII#Character_order). That said, if you want inputs to appear in a specific order, you need to name the inputs according to the ASCII-code order. For example, if you have three inputs: Root Zone Depth, Drainage, and Output. If you don't change their name, they will show in the model dialog in the following order:

- Drainage
- Output
- Root Zone Depth

So, to solve this, I recommend you to name them as: 1_RootZoneDepth, 2_Drainage, 3_Output. That is, add a number and underscore, e.g., "1_", "2_", as **a prefix** in front of all inputs.

**Comment by Achidago** [comment]
It works! well understood
Thanks @chjch
